mantidimaging/gui/test/gui_system_base.py

```diff
@@ -56,7 +56,10 @@ def tearDown(self) -> None:
         QTest.qWait(SHORT_DELAY)
         self.assertDictEqual(self.main_window.presenter.model.datasets, {})
 
-        if leak_count := leak_tracker.count():
+        # if self._outcome.result._excinfo is None then there were no AssertionErrors during the test
+        test_error = self._outcome.result._excinfo  # type: ignore
+        if leak_tracker.check_leak_limit_exceeded(self.leak_count_limit) and test_error is None:
+            leak_count = leak_tracker.count()
```

Comment:
The condition could be something like
`if test_error is None and (leak_count := leak_tracker.count()) > self.leak_count_limit:`

That would save needing the function call, and also would save calling `count()` when not needed.